### PR TITLE
On TF, don't request counts in `unique` unless needed.

### PR DESCRIPTION
**Pull request by hertschuh**
With the TensorFlow backend, `unique` currently always computes counts even when not needed. This may have a performance impact.

Also, `tf.unique` is compilable on TPU vs. `tf.unique_with_counts` is not.

Additionally modify `axis` in place and rename `dim_size` to `axis_dim` for readability and consistency with the other ops.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/backend/tensorflow/numpy.py [hunk 1]
```diff
@@ -3894,15 +3894,21 @@ def unique(
...
         axis_to_use = tf.constant([dim], dtype=tf.int32)
-        y, inverse, counts = tf.raw_ops.UniqueWithCountsV2(
-            x=x, axis=axis_to_use, out_idx=tf.int32
-        )
+        if return_counts:
+            y, inverse, counts = tf.raw_ops.UniqueWithCountsV2(
+                x=x, axis=axis_to_use, out_idx=tf.int32
+            )
+        else:
+            y, inverse = tf.raw_ops.UniqueV2(x=x, axis=axis_to_use)
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/backend/tensorflow/numpy.py]: Made consistent

[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/backend/tensorflow/numpy.py]: Made consistent

**Review by gaga1313 — APPROVED** [review]
Thanks for the fix! LGTM
